### PR TITLE
refactor: simplify _match

### PR DESCRIPTION
**Pull request by roelofvandijk**
The checks are the same for `arg`, `dtype` and `uop`.

### CODE REVIEW

**Comment by roelofvandijk** [comment]
This class would be simpler if these three would have signatures like `Optional[Set[DType]]`, set with 1 element if only one.

**Comment by Qazalin** [comment]
can you do a [run_process_replay] in your head commit to check kernels haven't changed?

File: tinygrad/codegen/uops.py [hunk 1]
```diff
@@ -80,22 +80,13 @@ class UPat:
...
-  if pat.dtype is not None:
-    if isinstance(pat.dtype, set):
-      if uop.dtype not in pat.dtype: return False
-    elif uop.dtype != pat.dtype: return False
-  if pat.uop is not None:
-    if isinstance(pat.uop, set):
-      if uop.uop not in pat.uop: return False
-    elif uop.uop != pat.uop: return False
+  for p, u in [(pat.arg,uop.arg), (pat.dtype, uop.dtype), (pat.uop, uop.uop)]:
+      if p is not None:
```
Comment (geohot) [inline tinygrad/codegen/uops.py:84]: Wrong spacing. Also, have you benchmarked this?

Comment (roelofvandijk) [reply tinygrad/codegen/uops.py]: 1) How did neither `pre-commit` nor the CI linter error on this?
2) Yes, I benchmarked it with ` JIT=0 python3 -O examples/llama.py --gen 1 --prompt "Hello." --count 10 --temperature 0 --timing`. No discernible speed differences. There are some performance optimizations possible here, but I did not apply them to keep the logic identical.

Comment (roelofvandijk) [reply tinygrad/codegen/uops.py]: Ah, `pylint` is not checking this folder because there is no `__init__.py`.

**Comment by roelofvandijk** [comment]
Replaced by https://github.com/tinygrad/tinygrad/pull/4876

**Pull request closed without merging**